ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/FilterTypeValidator.java

```diff
@@ -53,9 +53,15 @@ public FilterTypeValidator(
   public void validateFilterExpression(final Expression exp) {
     final SqlType type = getExpressionReturnType(exp);
     if (!SqlTypes.BOOLEAN.equals(type)) {
-      throw new KsqlException("Type error in " + filterType.name() + " expression: "
+      throw new KsqlStatementException(
```

Comment:
You'll see a lot of these in this PR. Basically, the first message is the one that will be part of a stacktrace, and the second one is for the API response. They are subtlely different, since the goal is to report the same information to the API in the same way (to support existing tooling), while reporting enough information in the logs to support debuggability without printing query text unnecessarily.

For example, in this diff, the second message contians the expression itself, while the first only contains the type.